miio/integrations/chunmi/cooker_multi/cooker_multi.py

```diff
@@ -0,0 +1,400 @@
+        )
+        checksum = bytearray(2)
+        checksum[0] = (crc >> 8) & 0xFF
+        checksum[1] = crc & 0xFF
+        return checksum
+
+    def update_checksum(self):
+        self.checksum = self.calc_checksum()
+
+    def is_valid(self):
+        return len(self.profile_bytes) == 174 and self.checksum == self.calc_checksum()
+
+    def get_profile_hex(self):
+        return (self.profile_bytes + self.checksum).hex()
+
+
+class CookerStatus(DeviceStatus):
+    def __init__(self, data):
+        self.data = data
+
+    @property
+    @sensor("Operation Mode")
+    def mode(self) -> OperationMode:
+        """Current operation mode."""
+        return OperationMode(self.data["status"])
+
+    @property
+    @sensor("Menu ID")
+    def menu(self) -> str:
+        """Selected menu id."""
+        try:
+            return COOKING_MENUS[self.data["menu"]]
+        except KeyError:
+            return "Unknown menu"
+
+    @property
+    @sensor("Cooking stage")
+    def stage(self) -> str:
+        """Current stage if cooking."""
+        try:
+            return COOKING_STAGES[self.data["phase"]]["name"]
+        except KeyError:
+            return "Unknown stage"
+
+    @property
+    @sensor("Current temperature", unit="C")
+    def temperature(self) -> Optional[int]:
+        """Current temperature, if idle.
+
+        Example values: 29
+        """
+        return self.data["temp"]
+
+    @property
+    @sensor("Cooking finish time")
+    def ready_at(self) -> datetime.datetime:
+        """Remaining minutes of the cooking process."""
+        return (
+            datetime.datetime.now()
+            + datetime.timedelta(
+                0, int(self.data["t_left"] + (self.data["t_pre"] * 60))
+            )
+        ).replace(second=0, microsecond=0)
+
+    @property
+    @sensor("Cooking start delay")
+    def cooking_delayed(self) -> Optional[int]:
+        """Wait n minutes before cooking / scheduled cooking."""
+        delay = int(self.data["t_pre"])
+
+        if delay >= 0:
+            return delay
+
+        return None
+
+    @property
+    @sensor("Cooking duration")
+    def duration(self) -> int:
+        """Duration of the cooking process."""
+        return int(self.data["t_cook"])
+
+    @property
+    @sensor("Keep warm after cooking enabled")
+    def keep_warm(self) -> bool:
+        """Keep warm after cooking?"""
+        return self.data["akw"] == 1
+
+    @property
+    @sensor("Taste ID")
+    def taste(self) -> None:
+        """Taste id."""
+        return self.data["taste"]
+
+    @property
+    @sensor("Rice ID")
+    def rice(self) -> None:
+        """Rice id."""
+        return self.data["rice"]
+
+    @property
+    @sensor("Selected favorite recipe")
+    def favorite(self) -> None:
+        """Favored recipe id."""
+        return self.data["favs"]
+
+
+class MultiCooker(Device):
+    """Main class representing the multi cooker."""
+
+    _supported_models = [*MODEL_MULTI]
```

Comment:
```suggestion
    _supported_models = [MODEL_MULTI]
```
I think? Could you try `miiocli device info` on the device to see if it gets detected correctly?

Reply:
Correct. Fixed